test/core/end2end/goaway_server_test.cc

```diff
@@ -165,7 +165,8 @@ int main(int argc, char** argv) {
   gpr_mu_init(&g_mu);
   grpc_init();
   g_default_dns_resolver = grpc_core::GetDNSResolver();
-  grpc_core::SetDNSResolver(new TestDNSResolver());
+  auto* resolver = new TestDNSResolver();
```

Comment:
Out of curiosity, how is this change related to the rest of this PR?  Did this PR somehow expose this pre-existing bug?

If not closely related, maybe split this into a separate PR?

Reply:
it got tickled by this pr and this seems to resolve it